File: Difficulty: Easy/Armstrong Numbers/armstrong-numbers.py
class Solution:
    def armstrongNumber(self, n):

        digits = len(str(n))   # power depends on number of digits
        temp = n
        arm = 0

        while temp != 0:
            t = temp % 10
            arm += t ** digits   # core logic: digit^digits
            temp //= 10

        return arm == n   # compare with original

File: Difficulty: Easy/Armstrong Numbers/armstrong-numbers.cpp
// User function Template for C++
class Solution {
  public:
    bool armstrongNumber(int n) {
       int temp = n;
       int sum = 0;
       while(temp != 0){
           int t = temp %10;
           sum += pow(t,3);
           temp = temp/10;
       }
       
       if(sum == n) return true;
       return false;
        
    }
};